### DISCUSSION TITLE
Why does the extension think it's not "prepared"?

### DISCUSSION THREAD

**Discussion by mattieb** [mshibanami/redirect-web#18] [Questions] [answered]
I've been trying this excellent extension across my Mac, iPhone, and iPad (all are running current public betas), but there's a weird thing going on—

Every time I open the app, I get the prompt to "prepare the extension".

On my Mac I've

1. turned it on in Safari
2. clicked the toolbar icon and allowed it (just on YouTube.com for me; I have not tried to allow it on every site)
3. pulled up the sheet in Safari to make sure it's working

and that all works, and the extension is working correctly, but the app thinks it needs to be prepared yet.

Am I missing something?

**Comment by mshibanami** [comment, marked as answer]
Hi @mattieb, sorry for any inconvenience.

On macOS, the app determines the extension's capabilities using [Safari's API](https://developer.apple.com/documentation/safariservices/sfsafariextensionmanager/2122754-getstateofsafariextension). There's a possibility that this API is not working correctly. Such issues can often be resolved by simply restarting macOS. If that doesn't work, please ensure your Safari and macOS are updated to the latest versions. Additionally, if you have Safari Technology Preview installed, I recommend uninstalling it as a precaution.

Should these steps not resolve the issue, deleting the settings file at the following location might help:

```
/Users/<your user ID>/Library/Group Containers/group.io.github.mshibanami.RedirectWebForSafari/Library/Preferences/group.io.github.mshibanami.RedirectWebForSafari.plist
```
